src/DDTrace/Span.php

```diff
@@ -4,139 +4,71 @@
 
 use DDTrace\Integrations\Integration;
 use DDTrace\Contracts\Span as SpanInterface;
+use DDTrace\Data\Span as SpanData;
+
 use DDTrace\Contracts\SpanContext as SpanContextInterface;
 use DDTrace\Exceptions\InvalidSpanArgument;
+use DDTrace\SpanContext as SpanContext;
 use DDTrace\Http\Urls;
+use DDTrace\Processing\TraceAnalyticsProcessor;
 use Exception;
 use InvalidArgumentException;
 use Throwable;
 
-final class Span implements SpanInterface
+final class Span extends SpanData
 {
-    /**
-     * Operation Name is the name of the operation being measured. Some examples
-     * might be "http.handler", "fileserver.upload" or "video.decompress".
-     * Name should be set on every span.
-     *
-     * @var string
-     */
-    private $operationName;
-
-    /**
-     * @var SpanContextInterface
-     */
-    private $context;
-
-    /**
-     * Resource is a query to a service. A web application might use
-     * resources like "/user/{user_id}". A sql database might use resources
-     * like "select * from user where id = ?".
-     *
-     * You can track thousands of resources (not millions or billions) so
-     * prefer normalized resources like "/user/{id}" to "/user/123".
-     *
-     * Resources should only be set on an app's top level spans.
-     *
-     * @var string
-     */
-    private $resource;
-
-    /**
-     * Service is the name of the process doing a particular job. Some
-     * examples might be "user-database" or "datadog-web-app". Services
-     * will be inherited from parents, so only set this in your app's
-     * top level span.
-     *
-     * @var string
-     */
-    private $service;
-
-    /**
-     * Protocol associated with the span
-     *
-     * @var string|null
-     */
-    private $type;
-
-    /**
-     * @var int
-     */
-    private $startTime;
-
-    /**
-     * @var int|null
-     */
-    private $duration;
-
-    /**
-     * @var array
-     */
-    private $tags = [];
-
-    /**
-     * @var array
-     */
-    private $metrics = [];
-
-    /**
-     * @var bool
-     */
-    private $hasError = false;
-
-    /**
-     * @var Integration
-     */
-    private $integration = null;
-
-    /**
-     * @var bool Whether or not this trace can be even considered for trace analytics automatic configuration.
-     */
-    private $isTraceAnalyticsCandidate = false;
+    private static $METRIC_NAMES = [ Tag::ANALYTICS_KEY => true ];
```

Comment:
Just a nitpick. I know these kind of act more like constants, hence the name in ALL CAPS and snake_case, but I'd still have this follow how other class property names using camelCase. And I'd do the same for `$SPECIAL_TAGS` below. Again - that's just my opinion, you don't have to change anything. :)

Reply:
Conventions are there to be broken 🙈 
Yeah it was a `const` but 5.4 made me change it to this :(.

I'll follow your suggestion though